### PR TITLE
feat(spanner): add custom instance config to tests

### PR DESCRIPTION
**Pull request by mohanli-ml**
In spanner integration tests, an instance will be created before the test. Currently, the instance is created with the first instance config. However, for staging environment the instance config is shared among GCP projects and can run out of resources. So this PR tries to solve this issue by allowing users to custom the instance config by adding an environment variable GCLOUD_TESTS_GOLANG_SPANNER_INSTANCE_CONFIG.

### CODE REVIEW

File: spanner/integration_test.go [hunk 1]
```diff
@@ -63,6 +63,11 @@ var (
 	// by setting the environment variable GCLOUD_TESTS_GOLANG_SPANNER_HOST
 	spannerHost = getSpannerHost()
 
+	// instanceConfig specifies the instance config used to create a instance for testing.
```
Comment (hengfengli) [inline spanner/integration_test.go:66]: a -> an

File: spanner/integration_test.go [hunk 2]
```diff
@@ -262,6 +271,10 @@ func initIntegrationTests() (cleanup func()) {
 	if err != nil {
 		log.Fatalf("Cannot get any instance configurations.\nPlease make sure the Cloud Spanner API is enabled for the test project.\nGet error: %v", err)
 	}
+	configName := config.Name
+	if instanceConfig != "" {
```
Comment (hengfengli) [inline spanner/integration_test.go:275]: If `instanceConfig` is not empty, there is no need to call `ListInstanceConfigs`. 

```go
        var configName string
	if instanceConfig != "" {
		configName = fmt.Sprintf("projects/%s/instanceConfigs/%s", testProjectID, instanceConfig)
	} else {
	        configIterator := instanceAdmin.ListInstanceConfigs(ctx, &instancepb.ListInstanceConfigsRequest{
		        Parent: fmt.Sprintf("projects/%s", testProjectID),
	        })
	        config, err := configIterator.Next()
	        if err != nil {
		        log.Fatalf("Cannot get any instance configurations.\nPlease make sure the Cloud Spanner API is enabled for the test project.\nGet error: %v", err)
	        }
	        configName := config.Name
        }
```

File: spanner/integration_test.go (hunk 1 above)
Comment (mohanli-ml) [reply spanner/integration_test.go]: Done

File: spanner/integration_test.go (hunk 2 above)
Comment (mohanli-ml) [reply spanner/integration_test.go:263]: Added an if-else.

**Review by hengfengli — APPROVED** [review]
LGTM.

**Pull request merged**